### DISCUSSION TITLE
Server-side functions with non-empty response.

### DISCUSSION THREAD

**Discussion by artmelancholy** [leptos-rs/leptos#1442] [Q&A] [answered]
Hello! 

First of all, thank you for your project, it makes work with web so more comfortable!

While I was reading leptos book, it was so easy to make client-side rendering. But in the end I've met some problems with server-side functions. [In this chapter](https://leptos-rs.github.io/leptos/server/25_server_functions.html) server function looks very easy, but it isn't returning something from server and `spawn_local..` return `()`. But what if I need to ask server and receive some data back?

**Comment by gbj** [comment, marked as answer]
Look at the final section of the chapter you linked to, or at some of the examples in this repo that use server functions (like `todo_app_sqlite`):

> Because a server function is just a plain Rust async function, it integrates perfectly with the async Leptos primitives we discussed [earlier](https://leptos-rs.github.io/leptos/async/index.html). So you can easily integrate your server functions with the rest of your applications:
> - Create resources that call the server function to load data from the server
> - Read these resources under <Suspense/> or <Transition/> to enable streaming SSR and fallback states while data loads.

In other words, say you have a server function called `my_server_function`: just create a resource that calls it.
```rust
let async_data = create_resource(cx,
    || (),
    |_| my_server_function(),
);
```

**Reply by artmelancholy** [reply]
Yayks, looks like I missed this part:( Thank you for clue it helped me a lot!
